script.js

```diff
@@ -149,9 +149,63 @@ function addCheckboxes() {
 		checkbox = document.createElement('input');
 		checkbox.type = 'checkbox';
 		checkbox.value = sha;
+		checkbox.onclick = highlightRange;
 		checkbox.className = 'commit-compare-checkbox';
 		
 		listItem.appendChild(checkbox);
+
+		function highlightRange() {
+			var allCheckboxes = getAllCheckboxElements(),
```

Comment:
I also thought about this but I felt that this would be more safe. 

The event handler is attached within the loop when the checkboxes are created. https://github.com/hengun348/GitHub-compare-helper/pull/2/commits/e602f7f084fcd6e8ef7faf5a3e41f74b69ec702d#diff-9a9569e9d73f33740eada95275da7f30R141
So with your suggestion we would use a variable where we are not certain that it is holding all checkboxes, or even one at all. Where would we run the `getAllCheckboxElements()`? We can't really run it before we have created them. 

That said, I don't see that in practice someone actually would be able to fire off the event handler before all checkboxes are created but it feels a bit wrong to depend on a variable outside of this function.

I guess one option could be to add the event listeners (`onclick`) after we have created all checkboxes. Although that would need a second loop.